### DISCUSSION TITLE
Export to JSON and import an Identity Provider via the Admin Console enhancement idea

### DISCUSSION THREAD

**Discussion by rmoqpd5t** [keycloak/keycloak#13555] [Ideas]
Clients can be exported via the Admin Console's per-client "Export" button.  Clients can be created via an Import button in the Admin Console.  As far as I know, Identity Providers cannot be exported or imported via the Admin Console.  Exportation of Identity Providers would be helpful for operations and change management.  Tested Identity Providers could be migrated to a Production instance of Keycloak without having to manually re-configure the Identity Provider, its roles, its mappers, etc.  This amount of manual re-configuration can be labourious and error-prone.  Exportation of Identity Providers would likely mirror how Clients can be exported via the Admin Console's per-client "Export" button.    Importation of Identity Providers would likely mirror how Clients can be imported via the Admin Console Add Client screen's Import button.

**Comment by ssilvert** [comment]
You can export and import identity providers through the admin console.

In the old console, use Import and Export on the left menu.

In the new console, go to Realm settings and use the action menu in the upper righthand corner.

**Comment by rmoqpd5t** [comment]
Thank-you @ssilvert for the suggestions. To my way of thinking, partial export is not a practical way to export and import a particular Identity Provider.  I'm hoping for discussion on the ability to export a tested Identity Provider and import that Identity Provider configuration into a production environment.  The partial export includes most realm configuration including *all* Identity Providers and all mappers.  I don't see how this file could not be safely imported into a production environment.  Stripping the partial export file of everything but the desired Identity Provider seems very challenging.  Particularly that Provider's mappers which are in a different section of the file than the identity providers only associated via the Identity Provider's alias.

In my case when an Identity provider is ready to go into production, it and it's 20+ mappers must be re-created in production and thoroughly re-tested in case of error.  Importation would ease this problem.  There still may be details to change (e.g EntityID, certificates, etc.) in Production.  But this process of moving between environments could be made more reliable.

In 19.01's new console from the "Action" menu the only option I see from an Identity Provider is "Delete".  Is the ability to export under action in a more recent branch?

**Reply by PrasannaVenkadesh** [reply]
I couldn't believe a simple feature request to export and import Identity Provider is open for 4 years!
